### PR TITLE
FIX-#5625: Fix set_index with modin series.

### LINKED ISSUE #5625: BUG: df.set_index() cannot use Modin Series to set the index
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd
d = {'col1': [1, 2, 3], 'col2': [1, 2, 3], 'col3': [1, 2, 3]}
df = pd.DataFrame(data=d)
s = pd.Series(["a", "b", "c"])
df = df.set_index(s)
```

### Issue Description

Modin DataFrame cannot receive a Modin/ pandas Series as an argument using df.set_index(). This results in `TypeError: The parameter "keys" may be a column key, one-dimensional array, or a list containing only valid column keys and one-dimensional arrays.. Received column of type <class 'modin.pandas.series.Series'>`

### Expected Behavior

Screen Shot 2023-02-03 at 3 36 56 PM

### Error Logs

<details>

```python-traceback

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/frame.py in set_index(self, keys, drop, append, inplace, verify_integrity)
   6001                 try:
-> 6002                     found = col in self.columns
   6003                 except TypeError as err:

~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py in __contains__(self, key)
   5288         """
-> 5289         hash(key)
   5290         try:

TypeError: unhashable type: 'Series'

The above exception was the direct cause of the following exception:

TypeError                                 Traceback (most recent call last)
/var/folders/qj/jybppsbd2jl75s8y2q8s2xx80000gn/T/ipykernel_40368/2952750038.py in <module>
      2 df = pd.DataFrame(data=d)
      3 s = pd.Series(["a", "b", "c"])
----> 4 df = df.set_index(s)
      5 df

~/opt/anaconda3/lib/python3.9/site-packages/modin/logging/logger_decorator.py in run_and_log(*args, **kwargs)
    126             """
    127             if LogMode.get() == "disable":
--> 128                 return obj(*args, **kwargs)
    129 
    130             logger = get_logger()

~/opt/anaconda3/lib/python3.9/site-packages/modin/pandas/dataframe.py in set_index(self, keys, drop, append, inplace, verify_integrity)
   2091             # calculation so that it matches.
   2092             frame.index = (
-> 2093                 pandas.DataFrame(index=self.index)
   2094                 .set_index(keys, append=append, verify_integrity=verify_integrity)
   2095                 .index

~/opt/anaconda3/lib/python3.9/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    329                     stacklevel=find_stack_level(),
    330                 )
--> 331             return func(*args, **kwargs)
    332 
    333         # error: "Callable[[VarArg(Any), KwArg(Any)], Any]" has no

~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/frame.py in set_index(self, keys, drop, append, inplace, verify_integrity)
   6002                     found = col in self.columns
   6003                 except TypeError as err:
-> 6004                     raise TypeError(
   6005                         f"{err_msg}. Received column of type {type(col)}"
   6006                     ) from err

TypeError: The parameter "keys" may be a column key, one-dimensional array, or a list containing only valid column keys and one-dimensional arrays.. Received column of type <class 'modin.pandas.series.Series'>

```

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 9068fbcfd7d4c169018ae706d6c96edea6db0cd1
python           : 3.9.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.6.0
Version          : Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

Modin dependencies
------------------
modin            : 0.18.1
ray              : 2.2.0
dask             : 2021.10.0
distributed      : 2021.10.0
hdk              : None

pandas dependencies
-------------------
pandas           : 1.5.3
numpy            : 1.20.3
pytz             : 2021.3
dateutil         : 2.8.2
setuptools       : 58.0.4
pip              : 21.2.4
Cython           : 0.29.24
pytest           : 6.2.4
hypothesis       : None
sphinx           : 4.2.0
blosc            : None
feather          : None
xlsxwriter       : 3.0.1
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : 2.9.3
jinja2           : 2.11.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.2
brotli           : 
fastparquet      : None
fsspec           : 2021.08.1
gcsfs            : None
matplotlib       : 3.6.3
numba            : 0.54.1
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 6.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.1
snappy           : None
sqlalchemy       : 1.4.22
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : 1.3.0
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5625 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/test_series.py [hunk 1]
```diff
@@ -2204,7 +2204,6 @@ def test_loc(data):
     data = np.arange(100)
     modin_series = pd.Series(data, index=index).sort_index()
     pandas_series = pandas.Series(data, index=index).sort_index()
-    # Using 'fmt: skip' below as 'black' and 'flake8' can't agree on how this should be formatted
```
Comment (dchigarev) [inline modin/pandas/test/test_series.py:2207]: why removed?

File: modin/pandas/dataframe.py [hunk 2]
```diff
@@ -2080,7 +2080,9 @@ def set_index(
             else:
                 frame = self.copy()
             if not all(
-                isinstance(col, (pandas.Index, Series, np.ndarray, list, Iterator))
+                isinstance(
+                    col, (pandas.Index, pandas.Series, np.ndarray, list, Iterator)
+                )
                 for col in keys
             ):
```
Comment (dchigarev) [inline modin/pandas/dataframe.py:2087]: this check should work the same however looks clearer IMO
```python
if any(isinstance(col, Series) for col in keys):
```

Comment (dchigarev) [reply modin/pandas/dataframe.py:2087]: or we can even drop this whole if-statement and replace it with a [`try_cast_to_pandas`](https://github.com/modin-project/modin/blob/21ab814e2f9fd9e4874a036e2fa8e53208638614/modin/utils.py#L480) call:
```python
keys = try_cast_to_pandas(keys)
```

File: modin/pandas/test/test_series.py (hunk 1 above)
Comment (mvashishtha) [reply modin/pandas/test/test_series.py]: Hm, this diff is outdated. I had an older version of `black` that wanted to delete this line. This change should be gone now.

File: modin/pandas/dataframe.py (hunk 2 above)
Comment (mvashishtha) [reply modin/pandas/dataframe.py]: I switched to `try_cast_to_pandas`. Thanks for the suggestion.

**Review by dchigarev — APPROVED** [review]
LGTM

File: modin/pandas/test/dataframe/test_indexing.py [hunk 3]
```diff
@@ -294,16 +294,28 @@ def test_indexing_duplicate_axis(data):
...
+    ids=[
+        "non_existing_column",
+        "first_column_name",
+        "original_index",
+        "list_of_index_and_first_column_name",
+        "series_of_integers",
+    ],
+)
+def test_set_index(data, key_func):
+    eval_general(*create_test_dfs(data), lambda df: df.set_index(key_func(df)))
```
Comment (anmyachev) [inline modin/pandas/test/dataframe/test_indexing.py:318]: If we are not testing `drop==False` parameter, then let's start.

Comment (mvashishtha) [reply modin/pandas/test/dataframe/test_indexing.py]: Done.

**Review by anmyachev — APPROVED** [review]
LGTM!

**Pull request merged**